### ISSUE TITLE
No JSON object could be decoded

### ISSUE THREAD

**Issue by 123Venom** [jurialmunkey/plugin.video.themoviedb.helper#287]
I'm assuming the new `_meta` key has something wrong as this is now showing in kodi.log.  I do not use this key either.

2020-08-16 16:16:39.712 T:3932  NOTICE: [plugin.video.themoviedb.helper]
                                            No JSON object could be decoded

**Comment by jurialmunkey** [comment]
What's the context around this log msg? I haven't seen it before. Unlikely to be anything to do with the player, as all the log messages in that module are prefixed "Player -- ".  Also, the meta keys use `dumps()` which is *en*coding not *de*coding. This message would come from somewhere using `loads()`

There's five different places in the code that use loads:

1. Loading player json files
2. Reading the playerstring in the player monitor used to update Trakt watched status after watching
3. In the TMDb module when downloading their daily dump of all collections/studios/networks IDs
4. In Trakt module to load token from settings string
5. Loading a jsonrpc library request

Definitely not (5) as that's in a try/except block and log message is prefixed "TMDbHelper - JSONRPC Error"
Unlikely to be (4) as token string is only ever written using `dumps()` and you'd know because Trakt wouldn't work.
Unlikely to be (2) as it is also only ever written using `dumps()` and a blank string won't get loaded.
Possibly (3) but that will only happen whilst trying to browse one of the "All" sections - e.g. "TheMovieDb > All Studios" - and you'd know it was that because it would fail to load the folder.

So that leaves (1) -- the only real explanation is that you've got an incorrectly configured (or empty) player json file in your players folder (other possibility is file corruption preventing a file from being read, but that's unlikely as I'd expect an error message along those lines).

**Comment by 123Venom** [comment]
did a few test tonight on this and in two different addons the moment a video file is played the error appears.  I kept the kodi.log file open and kept refreshing it to keep a close eye.  Here's a little more round it but not sure it's going to tell us more.

```2020-08-17 01:11:43.107 T:7496  NOTICE: Creating audio stream (codec id: 86060, channels: 8, sample rate: 48000, no pass-through)
2020-08-17 01:11:43.260 T:9528  NOTICE: DXVA::CProcessorHD::InitProcessor: supported deinterlace methods: blend:yes, bob:yes, adaptive:yes, mocomp:no.
2020-08-17 01:11:43.279 T:9528 WARNING: CWinRenderer::UpdateVideoFilter: chosen scaling method 1 is not supported by renderer
2020-08-17 01:11:43.347 T:3932  NOTICE: [plugin.video.themoviedb.helper]
                                            No JSON object could be decoded
2020-08-17 01:11:43.527 T:4436  NOTICE: CDVDVideoCodecFFmpeg::CDropControl: calculated diff time: 41727```

these 2 cases were done without calling up the CM and using anything TMDbHelper related, less the player files are still evaluated?

**Comment by jurialmunkey** [comment]
If it's happening without using TMDbHelper then it has to be (2) the Player Monitor attempting to load the playerstring, as that's the only thing that runs in the background that uses loads.

You can test if that's the issue by commenting out these two lines in service.py
https://github.com/jurialmunkey/plugin.video.themoviedb.helper/blob/master/resources/lib/service.py#L494-L495
``` python
# self.playerstring = _homewindow.getProperty('TMDbHelper.PlayerInfoString')
# self.playerstring = loads(self.playerstring) if self.playerstring else None
```
And replacing with `self.playerstring = None` and that should mean the log message stops.

If it is that, then it's pretty harmless - it's not throwing an exception, so nothing is breaking and having an empty json object there is no big deal because it's just going to be an empty dict. Just weird that you're even getting a message at all because it shouldn't load an empty string and the string is only written to when playing via TMDbHelper (so it should be empty otherwise).

The only other thing I can think of is that it's just the service monitor getting an empty or malformed response from the API (I assume the requests module uses loads to return a json object). It's unusual though because usually you'll get a server error code along with that sort of thing (e.g. a 404 or 500). Either way, probably pretty harmless.

**Comment by 123Venom** [comment]
ya it happens without using TMDbHelper so I figured it had to be service related and my hunch landed me right on those 2 lines you just showed me...lol.  Seemed like something from a player file was causing an issue there so to rule out `ME` being the cause for my own piece of mind I disabled my own addon, moved my own player files to my desktop and restarted kodi.  Still got the log error and even tried yet another new addon.  So I was confident I can rule out my player files.  Then I tried to comment out line 495 and 495 as advised, being a service change I restarted.  I was shocked to play a video and the moment playback began same log error surfaced again.

``2020-08-17 03:51:48.540 T:12128  NOTICE: VideoPlayer::OpenFile: plugin://plugin.video.seren/?action=getSources&actionArgs=%7B%22item_type%22%3A%20%22movie%22%2C%20%22trakt_id%22%3A%20481%7D
2020-08-17 03:51:48.989 T:16060  NOTICE: Creating InputStream
2020-08-17 03:51:49.669 T:16060  NOTICE: Creating Demuxer
2020-08-17 03:51:51.621 T:16060  NOTICE: Opening stream: 0 source: 256
2020-08-17 03:51:51.622 T:16060  NOTICE: Creating video codec with codec id: 173
2020-08-17 03:51:51.622 T:16060  NOTICE: CDVDVideoCodecFFmpeg::Open() Using codec: HEVC (High Efficiency Video Coding)
2020-08-17 03:51:51.622 T:16060  NOTICE: Creating video thread
2020-08-17 03:51:51.622 T:15876  NOTICE: running thread: video_thread
2020-08-17 03:51:51.622 T:16060  NOTICE: Opening stream: 1 source: 256
2020-08-17 03:51:51.622 T:16060  NOTICE: Finding audio codec for: 86060
2020-08-17 03:51:51.622 T:16060  NOTICE: CDVDAudioCodecFFmpeg::Open() Successful opened audio decoder truehd
2020-08-17 03:51:51.623 T:16060  NOTICE: Creating audio thread
2020-08-17 03:51:51.623 T:16060  NOTICE: Opening stream: 4 source: 256
2020-08-17 03:51:51.623 T:544  NOTICE: running thread: CVideoPlayerAudio::Process()
2020-08-17 03:51:51.633 T:544  NOTICE: Creating audio stream (codec id: 86060, channels: 8, sample rate: 48000, no pass-through)
2020-08-17 03:51:51.737 T:12128  NOTICE: DXVA::CProcessorHD::InitProcessor: supported deinterlace methods: blend:yes, bob:yes, adaptive:yes, mocomp:no.
2020-08-17 03:51:51.806 T:12128 WARNING: CWinRenderer::UpdateVideoFilter: chosen scaling method 1 is not supported by renderer
2020-08-17 03:51:52.005 T:15876  NOTICE: CDVDVideoCodecFFmpeg::CDropControl: calculated diff time: 41727
2020-08-17 03:51:52.288 T:18900  NOTICE: [plugin.video.themoviedb.helper]
                                            No JSON object could be decoded
```
